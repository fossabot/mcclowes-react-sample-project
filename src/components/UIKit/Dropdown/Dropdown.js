import React from "react";
import {
  DropdownWrapper,
  DropdownPlaceholder,
  DropdownOptions,
  DropdownOption
} from "./csx";
import Icon from "../Icon";

const Dropdown = props => {
  const { placeholder, options, open, setOpen, value, setValue } = props;

  return (
    <DropdownWrapper onClick={() => setOpen(!open)}>
      <DropdownPlaceholder value={value}>
        {value || placeholder}

        <Icon icon="TICK" />
      </DropdownPlaceholder>

      {options && (
        <DropdownOptions open={open}>
          {options.map((option, i) => {
            return (
              <DropdownOption
                key={i}
                onClick={() => setValue(option)}
                selected={value === option}
              >
                {option}
              </DropdownOption>
            );
          })}
        </DropdownOptions>
      )}
    </DropdownWrapper>
  );
};

export default Dropdown;
