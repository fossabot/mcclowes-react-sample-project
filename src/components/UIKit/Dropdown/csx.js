import styled from "styled-components";

export const DropdownPlaceholder = styled.div`
  padding: 0.25em 0.5em;
  display: flex;
  justify-content: space-between;
  flex-direction: row;
  align-items: center;

  ${props => (props.value ? "" : "color: #aaa;")};

  svg {
    fill: #aaa;
  }
`;

export const DropdownOptions = styled.div`
  ${props => (props.open ? "border-top: 1px solid #aaa;" : "display:none;")};
`;

export const DropdownOption = styled.div`
  padding: 0.25em 0.5em;

  &:hover {
    background-color: #eee;
  }

  &:active {
    background-color: #ddd;
  }
`;

export const DropdownWrapper = styled.div`
  border: 1px solid #aaa;
  border-radius: 3px;
  cursor: pointer;
`;
