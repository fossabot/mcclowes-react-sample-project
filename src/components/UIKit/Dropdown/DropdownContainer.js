import React, { useState } from "react";
import Dropdown from "./Dropdown";

const DropdownContainer = props => {
  const [open, setOpen] = useState(false);
  const [value, setValue] = useState(undefined);

  return (
    <Dropdown
      value={value}
      setValue={value => {
        setValue(value);
        props.callback(value);
      }}
      open={open}
      setOpen={setOpen}
      {...props}
    />
  );
};

export default DropdownContainer;
