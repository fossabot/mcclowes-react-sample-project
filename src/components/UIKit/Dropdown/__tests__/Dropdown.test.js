import Dropdown from "../DropdownContainer";
import React from "react";
import { mount } from "enzyme";

describe("Dropdown", () => {
  it("renders component", () => {
    const wrapper = mount(<Dropdown placeholder="Default Dropdown" />);

    expect(wrapper).toMatchSnapshot();
  });

  it("renders component with options", () => {
    const wrapper = mount(
      <Dropdown placeholder="Default Dropdown" options={[1, 2, 3, 4, 5]} />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it("open", () => {
    const wrapper = mount(<Dropdown placeholder="Default Dropdown" />);

    wrapper.simulate("click");

    expect(wrapper).toMatchSnapshot();
  });
});
