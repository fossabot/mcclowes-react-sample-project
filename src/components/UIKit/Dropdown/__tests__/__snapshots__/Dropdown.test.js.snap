// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown open 1`] = `
<DropdownContainer
  placeholder="Default Dropdown"
>
  <Dropdown
    open={true}
    placeholder="Default Dropdown"
    setOpen={[Function]}
    setValue={[Function]}
  >
    <styled.div
      onClick={[Function]}
    >
      <StyledComponent
        forwardedComponent={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-bxivhb",
              "isStatic": false,
              "lastClassName": "cZYlFj",
              "rules": Array [
                "
  border: 1px solid #aaa;
  border-radius: 3px;
  cursor: pointer;
",
              ],
            },
            "displayName": "styled.div",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-bxivhb",
            "target": "div",
            "toString": [Function],
            "usesTheme": false,
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          }
        }
        forwardedRef={null}
        onClick={[Function]}
      >
        <div
          className="sc-bxivhb cZYlFj"
          onClick={[Function]}
        >
          <styled.div>
            <StyledComponent
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-bdVaJa",
                    "isStatic": false,
                    "lastClassName": "kFYmqX",
                    "rules": Array [
                      "
  padding: 0.25em 0.5em;
  display: flex;
  justify-content: space-between;
  flex-direction: row;
  align-items: center;

  ",
                      [Function],
                      ";

  svg {
    fill: #aaa;
  }
",
                    ],
                  },
                  "displayName": "styled.div",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-bdVaJa",
                  "target": "div",
                  "toString": [Function],
                  "usesTheme": false,
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
            >
              <div
                className="sc-bdVaJa kFYmqX"
              >
                Default Dropdown
                <Icon
                  icon="TICK"
                >
                  <styled.svg
                    focusable="false"
                    theme={
                      Object {
                        "colors": Object {
                          "black": "#333",
                        },
                      }
                    }
                    viewBox="0 0 32 32"
                  >
                    <StyledComponent
                      focusable="false"
                      forwardedComponent={
                        Object {
                          "$$typeof": Symbol(react.forward_ref),
                          "_foldedDefaultProps": Object {
                            "theme": Object {
                              "colors": Object {
                                "black": "#333",
                              },
                            },
                          },
                          "attrs": Array [],
                          "componentStyle": ComponentStyle {
                            "componentId": "sc-ifAKCX",
                            "isStatic": false,
                            "lastClassName": "lgTGPa",
                            "rules": Array [
                              "
  fill: ",
                              [Function],
                              ";
  width: ",
                              [Function],
                              ";
  height: ",
                              [Function],
                              ";
",
                            ],
                          },
                          "displayName": "styled.svg",
                          "foldedComponentIds": Array [],
                          "render": [Function],
                          "styledComponentId": "sc-ifAKCX",
                          "target": "svg",
                          "toString": [Function],
                          "usesTheme": true,
                          "warnTooManyClasses": [Function],
                          "withComponent": [Function],
                        }
                      }
                      forwardedRef={null}
                      theme={
                        Object {
                          "colors": Object {
                            "black": "#333",
                          },
                        }
                      }
                      viewBox="0 0 32 32"
                    >
                      <svg
                        className="sc-ifAKCX lgTGPa"
                        focusable="false"
                        viewBox="0 0 32 32"
                      >
                        <path
                          d="M16.35,21.34L16.35,21.34l-2.83,2.83l-2.83-2.83l-6.36-6.36l2.83-2.83l6.36,6.36L24.84,7.2l2.83,2.83L16.35,21.34z"
                        />
                      </svg>
                    </StyledComponent>
                  </styled.svg>
                </Icon>
              </div>
            </StyledComponent>
          </styled.div>
        </div>
      </StyledComponent>
    </styled.div>
  </Dropdown>
</DropdownContainer>
`;

exports[`Dropdown renders component 1`] = `
<DropdownContainer
  placeholder="Default Dropdown"
>
  <Dropdown
    open={false}
    placeholder="Default Dropdown"
    setOpen={[Function]}
    setValue={[Function]}
  >
    <styled.div
      onClick={[Function]}
    >
      <StyledComponent
        forwardedComponent={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-bxivhb",
              "isStatic": false,
              "lastClassName": "cZYlFj",
              "rules": Array [
                "
  border: 1px solid #aaa;
  border-radius: 3px;
  cursor: pointer;
",
              ],
            },
            "displayName": "styled.div",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-bxivhb",
            "target": "div",
            "toString": [Function],
            "usesTheme": false,
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          }
        }
        forwardedRef={null}
        onClick={[Function]}
      >
        <div
          className="sc-bxivhb cZYlFj"
          onClick={[Function]}
        >
          <styled.div>
            <StyledComponent
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-bdVaJa",
                    "isStatic": false,
                    "lastClassName": "kFYmqX",
                    "rules": Array [
                      "
  padding: 0.25em 0.5em;
  display: flex;
  justify-content: space-between;
  flex-direction: row;
  align-items: center;

  ",
                      [Function],
                      ";

  svg {
    fill: #aaa;
  }
",
                    ],
                  },
                  "displayName": "styled.div",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-bdVaJa",
                  "target": "div",
                  "toString": [Function],
                  "usesTheme": false,
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
            >
              <div
                className="sc-bdVaJa kFYmqX"
              >
                Default Dropdown
                <Icon
                  icon="TICK"
                >
                  <styled.svg
                    focusable="false"
                    theme={
                      Object {
                        "colors": Object {
                          "black": "#333",
                        },
                      }
                    }
                    viewBox="0 0 32 32"
                  >
                    <StyledComponent
                      focusable="false"
                      forwardedComponent={
                        Object {
                          "$$typeof": Symbol(react.forward_ref),
                          "_foldedDefaultProps": Object {
                            "theme": Object {
                              "colors": Object {
                                "black": "#333",
                              },
                            },
                          },
                          "attrs": Array [],
                          "componentStyle": ComponentStyle {
                            "componentId": "sc-ifAKCX",
                            "isStatic": false,
                            "lastClassName": "lgTGPa",
                            "rules": Array [
                              "
  fill: ",
                              [Function],
                              ";
  width: ",
                              [Function],
                              ";
  height: ",
                              [Function],
                              ";
",
                            ],
                          },
                          "displayName": "styled.svg",
                          "foldedComponentIds": Array [],
                          "render": [Function],
                          "styledComponentId": "sc-ifAKCX",
                          "target": "svg",
                          "toString": [Function],
                          "usesTheme": true,
                          "warnTooManyClasses": [Function],
                          "withComponent": [Function],
                        }
                      }
                      forwardedRef={null}
                      theme={
                        Object {
                          "colors": Object {
                            "black": "#333",
                          },
                        }
                      }
                      viewBox="0 0 32 32"
                    >
                      <svg
                        className="sc-ifAKCX lgTGPa"
                        focusable="false"
                        viewBox="0 0 32 32"
                      >
                        <path
                          d="M16.35,21.34L16.35,21.34l-2.83,2.83l-2.83-2.83l-6.36-6.36l2.83-2.83l6.36,6.36L24.84,7.2l2.83,2.83L16.35,21.34z"
                        />
                      </svg>
                    </StyledComponent>
                  </styled.svg>
                </Icon>
              </div>
            </StyledComponent>
          </styled.div>
        </div>
      </StyledComponent>
    </styled.div>
  </Dropdown>
</DropdownContainer>
`;

exports[`Dropdown renders component with options 1`] = `
<DropdownContainer
  options={
    Array [
      1,
      2,
      3,
      4,
      5,
    ]
  }
  placeholder="Default Dropdown"
>
  <Dropdown
    open={false}
    options={
      Array [
        1,
        2,
        3,
        4,
        5,
      ]
    }
    placeholder="Default Dropdown"
    setOpen={[Function]}
    setValue={[Function]}
  >
    <styled.div
      onClick={[Function]}
    >
      <StyledComponent
        forwardedComponent={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-bxivhb",
              "isStatic": false,
              "lastClassName": "cZYlFj",
              "rules": Array [
                "
  border: 1px solid #aaa;
  border-radius: 3px;
  cursor: pointer;
",
              ],
            },
            "displayName": "styled.div",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-bxivhb",
            "target": "div",
            "toString": [Function],
            "usesTheme": false,
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          }
        }
        forwardedRef={null}
        onClick={[Function]}
      >
        <div
          className="sc-bxivhb cZYlFj"
          onClick={[Function]}
        >
          <styled.div>
            <StyledComponent
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-bdVaJa",
                    "isStatic": false,
                    "lastClassName": "kFYmqX",
                    "rules": Array [
                      "
  padding: 0.25em 0.5em;
  display: flex;
  justify-content: space-between;
  flex-direction: row;
  align-items: center;

  ",
                      [Function],
                      ";

  svg {
    fill: #aaa;
  }
",
                    ],
                  },
                  "displayName": "styled.div",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-bdVaJa",
                  "target": "div",
                  "toString": [Function],
                  "usesTheme": false,
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
            >
              <div
                className="sc-bdVaJa kFYmqX"
              >
                Default Dropdown
                <Icon
                  icon="TICK"
                >
                  <styled.svg
                    focusable="false"
                    theme={
                      Object {
                        "colors": Object {
                          "black": "#333",
                        },
                      }
                    }
                    viewBox="0 0 32 32"
                  >
                    <StyledComponent
                      focusable="false"
                      forwardedComponent={
                        Object {
                          "$$typeof": Symbol(react.forward_ref),
                          "_foldedDefaultProps": Object {
                            "theme": Object {
                              "colors": Object {
                                "black": "#333",
                              },
                            },
                          },
                          "attrs": Array [],
                          "componentStyle": ComponentStyle {
                            "componentId": "sc-ifAKCX",
                            "isStatic": false,
                            "lastClassName": "lgTGPa",
                            "rules": Array [
                              "
  fill: ",
                              [Function],
                              ";
  width: ",
                              [Function],
                              ";
  height: ",
                              [Function],
                              ";
",
                            ],
                          },
                          "displayName": "styled.svg",
                          "foldedComponentIds": Array [],
                          "render": [Function],
                          "styledComponentId": "sc-ifAKCX",
                          "target": "svg",
                          "toString": [Function],
                          "usesTheme": true,
                          "warnTooManyClasses": [Function],
                          "withComponent": [Function],
                        }
                      }
                      forwardedRef={null}
                      theme={
                        Object {
                          "colors": Object {
                            "black": "#333",
                          },
                        }
                      }
                      viewBox="0 0 32 32"
                    >
                      <svg
                        className="sc-ifAKCX lgTGPa"
                        focusable="false"
                        viewBox="0 0 32 32"
                      >
                        <path
                          d="M16.35,21.34L16.35,21.34l-2.83,2.83l-2.83-2.83l-6.36-6.36l2.83-2.83l6.36,6.36L24.84,7.2l2.83,2.83L16.35,21.34z"
                        />
                      </svg>
                    </StyledComponent>
                  </styled.svg>
                </Icon>
              </div>
            </StyledComponent>
          </styled.div>
          <styled.div
            open={false}
          >
            <StyledComponent
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-bwzfXH",
                    "isStatic": false,
                    "lastClassName": "kIdcZD",
                    "rules": Array [
                      "
  ",
                      [Function],
                      ";
",
                    ],
                  },
                  "displayName": "styled.div",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-bwzfXH",
                  "target": "div",
                  "toString": [Function],
                  "usesTheme": false,
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
              open={false}
            >
              <div
                className="sc-bwzfXH kIdcZD"
                open={false}
              >
                <styled.div
                  key="0"
                  onClick={[Function]}
                  selected={false}
                >
                  <StyledComponent
                    forwardedComponent={
                      Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "attrs": Array [],
                        "componentStyle": ComponentStyle {
                          "componentId": "sc-htpNat",
                          "isStatic": false,
                          "lastClassName": "jQWAJu",
                          "rules": Array [
                            "
  padding: 0.25em 0.5em;
",
                          ],
                        },
                        "displayName": "styled.div",
                        "foldedComponentIds": Array [],
                        "render": [Function],
                        "styledComponentId": "sc-htpNat",
                        "target": "div",
                        "toString": [Function],
                        "usesTheme": false,
                        "warnTooManyClasses": [Function],
                        "withComponent": [Function],
                      }
                    }
                    forwardedRef={null}
                    onClick={[Function]}
                    selected={false}
                  >
                    <div
                      className="sc-htpNat jQWAJu"
                      onClick={[Function]}
                      selected={false}
                    >
                      1
                    </div>
                  </StyledComponent>
                </styled.div>
                <styled.div
                  key="1"
                  onClick={[Function]}
                  selected={false}
                >
                  <StyledComponent
                    forwardedComponent={
                      Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "attrs": Array [],
                        "componentStyle": ComponentStyle {
                          "componentId": "sc-htpNat",
                          "isStatic": false,
                          "lastClassName": "jQWAJu",
                          "rules": Array [
                            "
  padding: 0.25em 0.5em;
",
                          ],
                        },
                        "displayName": "styled.div",
                        "foldedComponentIds": Array [],
                        "render": [Function],
                        "styledComponentId": "sc-htpNat",
                        "target": "div",
                        "toString": [Function],
                        "usesTheme": false,
                        "warnTooManyClasses": [Function],
                        "withComponent": [Function],
                      }
                    }
                    forwardedRef={null}
                    onClick={[Function]}
                    selected={false}
                  >
                    <div
                      className="sc-htpNat jQWAJu"
                      onClick={[Function]}
                      selected={false}
                    >
                      2
                    </div>
                  </StyledComponent>
                </styled.div>
                <styled.div
                  key="2"
                  onClick={[Function]}
                  selected={false}
                >
                  <StyledComponent
                    forwardedComponent={
                      Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "attrs": Array [],
                        "componentStyle": ComponentStyle {
                          "componentId": "sc-htpNat",
                          "isStatic": false,
                          "lastClassName": "jQWAJu",
                          "rules": Array [
                            "
  padding: 0.25em 0.5em;
",
                          ],
                        },
                        "displayName": "styled.div",
                        "foldedComponentIds": Array [],
                        "render": [Function],
                        "styledComponentId": "sc-htpNat",
                        "target": "div",
                        "toString": [Function],
                        "usesTheme": false,
                        "warnTooManyClasses": [Function],
                        "withComponent": [Function],
                      }
                    }
                    forwardedRef={null}
                    onClick={[Function]}
                    selected={false}
                  >
                    <div
                      className="sc-htpNat jQWAJu"
                      onClick={[Function]}
                      selected={false}
                    >
                      3
                    </div>
                  </StyledComponent>
                </styled.div>
                <styled.div
                  key="3"
                  onClick={[Function]}
                  selected={false}
                >
                  <StyledComponent
                    forwardedComponent={
                      Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "attrs": Array [],
                        "componentStyle": ComponentStyle {
                          "componentId": "sc-htpNat",
                          "isStatic": false,
                          "lastClassName": "jQWAJu",
                          "rules": Array [
                            "
  padding: 0.25em 0.5em;
",
                          ],
                        },
                        "displayName": "styled.div",
                        "foldedComponentIds": Array [],
                        "render": [Function],
                        "styledComponentId": "sc-htpNat",
                        "target": "div",
                        "toString": [Function],
                        "usesTheme": false,
                        "warnTooManyClasses": [Function],
                        "withComponent": [Function],
                      }
                    }
                    forwardedRef={null}
                    onClick={[Function]}
                    selected={false}
                  >
                    <div
                      className="sc-htpNat jQWAJu"
                      onClick={[Function]}
                      selected={false}
                    >
                      4
                    </div>
                  </StyledComponent>
                </styled.div>
                <styled.div
                  key="4"
                  onClick={[Function]}
                  selected={false}
                >
                  <StyledComponent
                    forwardedComponent={
                      Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "attrs": Array [],
                        "componentStyle": ComponentStyle {
                          "componentId": "sc-htpNat",
                          "isStatic": false,
                          "lastClassName": "jQWAJu",
                          "rules": Array [
                            "
  padding: 0.25em 0.5em;
",
                          ],
                        },
                        "displayName": "styled.div",
                        "foldedComponentIds": Array [],
                        "render": [Function],
                        "styledComponentId": "sc-htpNat",
                        "target": "div",
                        "toString": [Function],
                        "usesTheme": false,
                        "warnTooManyClasses": [Function],
                        "withComponent": [Function],
                      }
                    }
                    forwardedRef={null}
                    onClick={[Function]}
                    selected={false}
                  >
                    <div
                      className="sc-htpNat jQWAJu"
                      onClick={[Function]}
                      selected={false}
                    >
                      5
                    </div>
                  </StyledComponent>
                </styled.div>
              </div>
            </StyledComponent>
          </styled.div>
        </div>
      </StyledComponent>
    </styled.div>
  </Dropdown>
</DropdownContainer>
`;
